### DISCUSSION TITLE
@datorama/akita-ng-effects for TypeScript 2.7.2 or a workaround with ts-action along with @datorama/akita@1.24.3

### DISCUSSION THREAD

**Discussion by vegeta03** [salesforce/akita#655] [Q&A]
I'm submitting a support/question issue.
Is there a way to use `@datorama/akita-ng-effects` with Angular 6.1.0 with TypeScript 2.7.2,  Upgrade is not possible because the project is huge with several dependencies.
I found out `@datorama/akita@1.24.3` is compatible for my situation, but I also need `@datorama/akita-ng-effects`
Or is there a workaround for getting functionality provided by  `@datorama/akita-ng-effects` using ts-action lib or any way?

[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[x] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

```
ERROR in node_modules/ts-action/action.d.ts(3,52): error TS1005: ';' expected.
node_modules/ts-action/action.d.ts(5,3): error TS1128: Declaration or statement expected.
node_modules/ts-action/action.d.ts(6,12): error TS1005: ',' expected.
node_modules/ts-action/action.d.ts(8,37): error TS1005: ';' expected.
node_modules/ts-action/action.d.ts(10,3): error TS1128: Declaration or statement expected.
node_modules/ts-action/action.d.ts(10,66): error TS1005: ';' expected.
node_modules/ts-action/action.d.ts(12,39): error TS1005: ';' expected.
node_modules/ts-action/action.d.ts(12,67): error TS1005: ';' expected.
node_modules/ts-action/action.d.ts(12,76): error TS1005: ';' expected.
node_modules/ts-action/action.d.ts(12,79): error TS1109: Expression expected.
node_modules/ts-action/action.d.ts(12,105): error TS1005: '(' expected.
node_modules/ts-action/action.d.ts(12,112): error TS1005: ')' expected.
node_modules/ts-action/action.d.ts(58,39): error TS1005: ';' expected.
node_modules/ts-action/action.d.ts(59,26): error TS1005: '(' expected.
node_modules/ts-action/action.d.ts(60,32): error TS1005: ',' expected.
node_modules/ts-action/action.d.ts(62,44): error TS1005: ';' expected.
node_modules/ts-action/action.d.ts(63,17): error TS1005: ']' expected.
node_modules/ts-action/action.d.ts(63,18): error TS1005: ',' expected.
node_modules/ts-action/action.d.ts(63,19): error TS1136: Property assignment expected.
node_modules/ts-action/action.d.ts(63,40): error TS1005: ',' expected.
node_modules/ts-action/action.d.ts(63,43): error TS1005: ':' expected.
node_modules/ts-action/action.d.ts(63,44): error TS1109: Expression expected.
node_modules/ts-action/action.d.ts(65,35): error TS1005: '(' expected.
node_modules/ts-action/action.d.ts(66,41): error TS1005: ',' expected.
node_modules/ts-action/types.d.ts(5,43): error TS1005: ';' expected.
node_modules/ts-action/types.d.ts(5,67): error TS1005: ',' expected.
```

## Expected behaviour

No typings issue.

## Minimal reproduction of the problem with instructions
Create an `Angular@6.1.0` project with the following dependency versions:  

```
"@angular/core": "^6.1.0",
"@datorama/akita": "1.24.3",
"@datorama/akita-ng-effects": "1.0.0",
```

## What is the motivation/use case for changing the behaviour?

Since the Angular and Typescript versions are older, the functionality provided by Akita is more helpful than any other state management library.

## Environment

```
"@angular/core": "^6.1.0",
"@datorama/akita": "1.24.3",
"@datorama/akita-ng-effects": "1.0.0",
```
 
For Tooling issues:
- Node version: 12
- Platform:  Windows 

Others:  
Stackblitz: https://stackblitz.com/github/vegeta03/immer-akita-ng-effects  
Github: https://github.com/vegeta03/immer-akita-ng-effects

**Comment by NetanelBasal** [comment]
@yackinn?

**Comment by yackinn** [comment]
ts-action and ts-action-operators depend on typescript 4.
We can't use typescript version 2 as ngcc won't support it. 
The only solution would be downgrading ts-action and ts-action-operators from version 11 to version 6.

I don't think it would be a good solution.
There are also several type errors when using ts-action version 6.

**Reply by vegeta03** [reply]
@NetanelBasal / @yackinn ,
Can you suggest a workaround to accomplish http calls on action dispatch just using `@datorama/akita@1.24.3` to update the store.

**Reply by yackinn** [reply]
If you want to stick to rxjs and observables you can write a simple observable/subscription service that publishes an event upon a certain action. You can do a a http call when the event has triggered.
A simpler approach is using a service that triggers your action and does a http call.

It depends on what you want to accomplish.

Just make sure not to get caught in spaghetti code when using services.
Stick to single responsibility methods.

**Reply by vegeta03** [reply]
@yackinn
What I am trying to get is to listen for actions dispatched.
Like we import and inject Actions and use ofType, to dispatch another action.

**Reply by yackinn** [reply]
How do you dispatch actions now?

Without using akita effects you can build your own observable service and publish events when calling methods from your service when you want to update the state.

**Reply by vegeta03** [reply]
Right now not using State management.
I'll try with Observable service.
Thank you.
